text/204-golang-bindings.md

```diff
@@ -66,8 +66,8 @@ package launchtype
 type LaunchType string
 
 const (
-    LaunchTypeEc2     LaunchType = "EC2"
-    LaunchTypeFargate LaunchType = "FARGATE"
+    LaunchTypeEC2     LaunchType = "EC2"
+    LaunchTypeFARGATE LaunchType = "FARGATE"
```

Comment:
Counter-argument here (just occurred to me) - is this going to look very awkward when the enum constant name includes `_` characters? This looks kinda "meh":

```go
LaunchTypePRETEND_THIS_EXISTS LaunchType = "PRETEND_THIS_EXISTS"
```

But on the other hand, we'd get into problems if we have the following (as the enum constant `LaunchTypePretendThisExists` and the class would end up getting the exact same name):

```ts
export enum LaunchType {
  PRETEND_THIS_EXISTS
}

export class LaunchTypePretendThisExists implements ILaunchType {
  //...
}
```

(Not that this problem doesn't also exist with the `SHOUT_CASED` names without `_`, too, e.g: `LaunchTypeEC2` class and `LaunchType.EC2` enum constant)

Reply:
Is there a naming convention for Go consts? We should follow it if there is one.

I am wondering if perhaps we can just block something like `LaunchTypePretendThisExists` at the `jsii` compiler level/

As for underscores, many of our enums have underscores in them, so I think this is actually a pretty major reason not to do that. Alternatively, maybe we can add an underscore between the enum name and the member (something like `LaunchType_PRETENT_THIS_EXISTS`).

Reply:
Could not find a convention for constant naming - and [effective go](https://golang.org/doc/effective_go#constants) actually has examples in varying case formats... 

I've updated the PR with an `_` including example, and standardised on `TypeName_MEMBER_NAME`.

Reply:
Sounds good. I think we can also standardize static methods/fields with an underscore between the type name and the member because we want them to be "enum-like". It's common to implement enums using public static fields.